### PR TITLE
Reload Configuration only when a pod matching the label set in mounted-file is created or deleted

### PR DESCRIPTION
**Pull request by huskykurt**
According to the issue comments ([comment1](https://github.com/vmware/kube-fluentd-operator/issues/289#issuecomment-985904900), [comment2](https://github.com/vmware/kube-fluentd-operator/issues/289#issuecomment-1141027421)) the existing solution caused too many configuration reloads.
This is because the configuration reload occurred whenever every pod was created (or deleted). So it is not affordable for a large cluster.
So, make the configuration reload only when a pod with the label set in the mounted-file is created (or deleted).

### CODE REVIEW

**Comment by huskykurt** [comment]
Hello @javiercri. How do I get a review for this PR? I looked at the [CLA FAQ](https://cla.vmware.com/faq) and they said that if I send a PR, I will receive a comment requesting the CLA, but there is no comment.

**Comment by javiercri** [comment]
First sorry for the late reply.
Could you fix the conflicts here @huskykurt, and I can approve it.
Thanks in advance!

**Review by javiercri — CHANGES_REQUESTED** [review]

File: config-reloader/util/util.go [hunk 1]
```diff
@@ -123,3 +130,63 @@ func TrimTrailingComment(line string) string {
...
+	}
+
+	if len(result) == 0 {
+		return nil, errors.New("at least one label must be given")
+	}
+
+	return result, nil
+}
+
+func Match(labels map[string]string, contLabels map[string]string, contName string) bool {
```
[the remark this answers is not in the document]
Comment (huskykurt) [reply config-reloader/util/util.go:185]: @javiercri I added a test for this function

**Review by javiercri — APPROVED** [review]
LGTM

**Pull request merged**